server-phoenix/lib/helios/widget.ex

```diff
@@ -1,21 +1,38 @@
 defmodule Helios.Widget do
   use Ecto.Schema
   import Ecto.Changeset
-  alias Helios.Widget
+  alias Helios.Location
+  import Ecto.Query
 
   @primary_key {:id, :id, autogenerate: true}
   schema "widgets" do
-    field :name, :string
-    field :enabled, :boolean
-    field :duration_seconds, :integer
-    field :position, :integer
-    field :created_at, :utc_datetime
-    field :updated_at, :utc_datetime
+    field :name, :string, presence: true
+    field :enabled, :boolean, presence: true
+    field :duration_seconds, :integer, presence: true
+    field :position, :integer, presence: true
+    field :created_at, :utc_datetime, presence: true
+    field :updated_at, :utc_datetime, presence: true
     field :start, :time
     field :stop, :time
     field :sidebar_text, :string
     field :show_weather, :boolean
 
-    belongs_to :location, Helios.Location
+    belongs_to :location, Location
+  end
+
+  def enabled(query) do
+    from q in query, where: q.enabled
+  end
+
+  def available(query, time) when is_binary(time) do
+    case DateTime.from_iso8601(time) do
+      {:ok, datetime, _} -> available(query, datetime)
+      _ -> {:error, "Datetime string is not in the iso format"}
+    end
+  end
+
+  def available(query, time) do
+    tod = DateTime.to_time(time)
+    from q in query, where: (q.start <= ^tod or is_nil(q.start)) and (q.stop >= ^tod or is_nil(q.stop))
```

Comment:
:thinking: I think the return values for overloaded functions with pattern matching should probably be the same. This one returns a query and the other might return a tuple or a query.